homeassistant/exceptions.py

```diff
@@ -64,6 +64,15 @@ def __str__(self) -> str:
             return self._message
 
         if not self.generate_message:
+            # Initialize self._message to the string repr of the class
+            # to prevent a recursive loop.
+            self._message = (
+                f"Parent class {self.__class__.__name__} is missing __str__ method"
+            )
+            # If the there is an other super class involved,
+            # we want to call its __str__ method.
+            # If the super().__str__ method is missing in the base_class
```

Comment:
All exceptions that inherit Exception have a `__str__` magic method. I'm not following what we mean here.